pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -735,27 +796,58 @@ installPhase:
 
 		default:
 			for _, installWorker := range pplnPhase.q {
+				appScope := getAppSrcScope(installWorker.afwConfig, installWorker.appSrcName)
+				if enterpriseApi.ScopeLocal != appScope {
+					scopedLog.Error(nil, "Install worker with non-local scope", "name", installWorker.cr.GetName(), "namespace", installWorker.cr.GetNamespace(), "pod name", installWorker.targetPodName, "App name", installWorker.appDeployInfo.AppName, "digest", installWorker.appDeployInfo.ObjectHash, "scope", appScope)
+					continue
+				}
+
 				phaseInfo := getPhaseInfoByPhaseType(installWorker, enterpriseApi.PhaseInstall)
 				if checkIfWorkerIsEligibleForRun(installWorker, phaseInfo, enterpriseApi.AppPkgInstallComplete) {
 					installWorker.waiter = &pplnPhase.workerWaiter
 					select {
 					case pplnPhase.msgChannel <- installWorker:
 						scopedLog.Info("Install worker got a run slot", "name", installWorker.cr.GetName(), "namespace", installWorker.cr.GetNamespace(), "pod name", installWorker.targetPodName, "App name", installWorker.appDeployInfo.AppName, "digest", installWorker.appDeployInfo.ObjectHash)
 						installWorker.isActive = true
+
+						// ToDo Start: sgontla: Fix the flow once CSPL-1449 is in place
+						var localInstallCtxt *localScopeInstallContext = &localScopeInstallContext{
+							worker: installWorker,
+							//sem:
+						}
+
+						//installWorker.waiter.Add(1)
+						localInstallCtxt.runPlaybook()
+						// just drain the channel for now, should handle in CSPL-1449
+						//<-pplnPhase.msgChannel
+						// ToDo End: sgontla: Fix the flow once CSPL-1449 is in place
 					default:
 						installWorker.waiter = nil
 					}
 				} else if phaseInfo.Status == enterpriseApi.AppPkgInstallComplete {
```

Comment:
That is a really good point. Now that, we are adding more status values. Instead of adding all those conditional checks as part of the `checkIfWorkerIsEligibleForRun`, you are right that moving to the end makes it simple, and that way we don't miss it in the future for any more conditions. I modify it.